src/src/PluginStructs/P036_data_struct.cpp

```diff
@@ -495,17 +490,21 @@ tFontSettings P036_data_struct::CalculateFontSettings(uint8_t lDefaultLines)
       }
     }
     if (iFontIndex < 0) {
-      // no font fits -> rduce number of lines per page
+      // no font fits -> reduce number of lines per page
       iLinesPerFrame--;
-      if (iLinesPerFrame == 0)
+      if (iLinesPerFrame == 0) {
+        // lines per frame is at minimum
         break;
+      }
     }
   }
   if (iFontIndex >= 0) {
     // font found -> calculate top position and space between lines
     iMaxHeightForFont = FontSizes[iFontIndex].Height * iLinesPerFrame;
-    if (iLinesPerFrame > 1)
+    if (iLinesPerFrame > 1) {
+      // more than one lines per frame -> calculate space inbetween
       result.Space = (iHeight-iMaxHeightForFont) / iLinesPerFrame;
+    }
     result.Top = (iHeight - (iMaxHeightForFont + (result.Space * (iLinesPerFrame-1)))) / 2;
```

Comment:
```c++
    result.Top = (iHeight - (iMaxHeightForFont + (result.Space * (iLinesPerFrame-1)))) / 2;
```
`iLinesPerFrame` can be 0 here, but it took me a while to realize `result.Space` is 0 at that point.
Maybe you can move the line `result.Space = 0;` to right above `if (iFontIndex >= 0) {` to make it more clear?
It is a prerequisite for this to work, so it should be close or else a reader ("me") will miss it later.

Reply:
Done, see my next commit.